fluent-syntax/src/parser.js

```diff
@@ -415,80 +415,83 @@ export default class FluentParser {
   }
 
   getPattern(ps) {
-    let buffer = '';
     const elements = [];
-    let firstLine = true;
+    ps.skipInlineWS();
 
-    if (this.withSpans) {
-      var spanStart = ps.getIndex();
+    // Special-case: trim leading whitespace and newlines.
+    if (ps.isPeekNextLinePattern()) {
+      ps.skipBlankLines();
+      ps.skipInlineWS();
```

Comment:
it would be nice if we could just peek to the first character that is non-WS and if it's a pattern start, skip to peek here, instead of resetting and skipping again, but since it's not a runtime parser, I don't think it's a blocker.